sdk/core/azure-json/src/main/java/com/azure/json/IOExceptionFunction.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.json;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+/**
+ * Equivalent in functionality to {@link Function} except that {@link #apply(Object)} is checked with
+ * {@link IOException}.
+ *
+ * @param <T> Input type of the function.
+ * @param <R> Output type of the function.
+ */
+@FunctionalInterface
+public interface IOExceptionFunction<T, R> {
```

Comment:
same here, do we really need to expose it?

Reply:
This is taken in a few `JsonReader` APIs and it greatly simplifies the need to handle try-catching in lambdas. Same goes for the similar interfaces that match Java interfaces but with checked exceptions.